quiz-bundle-spec.md

```diff
@@ -0,0 +1,91 @@
+# Qwiklabs Quiz Bundle Specification
+
+**Version 1**
+
+> This is a DRAFT document. We welcome feedback as this format evolves.
+
+## `qwiklabs.yaml` Structure
+
+Here's a sample `qwiklabs.yaml` file with all nested details removed to make it easier to see the general file structure.
+
+```yml
+entity_type: Quiz
+schema_version: 1
+
+default_locale: en
+
+title: Super Hard Quiz
+
+# The threshold grade that a student needs to achieve in order to count as "passing" the quiz.
+passing_percentage: 67
+
+# The "items" (i.e. questions) contained in this quiz.
+items: ...
+```
+
+Note that all of the localized content (stem, option titles, rationales, etc) are HTML content that may be displayed in various contexts. All of these chunks will be sanitized according to the restricted set in the [HTML spec](./html-spec.md).
+
+### Quiz attributes
+
+attribute          | required | type    | notes
+-------------------| -------- | ------- | -----------------------------------------
+title              | ✓        | string  | The authoring title of the quiz - not visible to learners.
+passing_percentage | ✓        | integer | The threshold grade that a student needs to achieve in order to count as "passing" the quiz.
+items              | ✓        | array   | An ordered array of `items` (see below for details) in this quiz - items will appear to students in this order
+
+### Default locale
+
+The quiz bundle must specify a `default_locale`. It corresponds to the locale that the quiz is authored in. Authoring tools can use this as a hint to notify localizers when content in the default locale is updated. Also, it provides a hint to the learner interface about which locale to display if an instruction/resource is not localized for the learner's current locale.
+
+### Items
+Items are polymorphic - i.e. there are several different _item types_ that are defined slightly differently. `items` is an array of dictionaries with appropriate attributes for the given `type`. The allowed values for `type` are:
+- `multiple-choice`
+- `multiple-select`
+- `true-false`
+
+#### multiple-choice Items
+
+A quiz item that has multiple options and one answer. One option is the _answer_, and the rest are _distractors_. There must be exactly one answer.
+
+attribute | required | type       | notes
+----------| -------- | -----------| -----------------------------------------
+id        | ✓        | string     | A unique identifier for this item
+type      | ✓        | string     | The item type, which is always `multiple-choice`
+stem      | ✓        | dictionary | A locale dictionary of the text that asks the question, such as "Which of the following is a color?"
```

Comment:
@rickcnagy @joncass - I would prefer this be a "stronger" type than just dictionary. We aren't accepting an arbitrary dictionary here. It's specifically a "LocaleDictionary" (not pushing this as the official name) which has a very specific structure.

It's the same reasoning for why Options was pulled out to a separate definition, and even before that you felt compelled to put the accepted Options properties in "nested" rows.

If you want to just leave it as `dictionary` for now, and we can update it all the references across all the documents after overhauling (and centralizing) the localization documentation, that's cool with me. However, I want to call out that generic `dictionary` and `array` should be avoided throughout this repo. We don't have any instance of "unspecified" / "blobby" input yet, and I suspect we never will.

Reply:
^^ I say we go for a unified "LocaleDictionary" type that's consistent across all of the specs (and then can share unbundling code as well!)